src/realm/object-store/impl/collection_change_builder.cpp

```diff
@@ -675,6 +714,12 @@ CollectionChangeSet CollectionChangeBuilder::finalize() &&
     // but we don't want inserts in the final modification set
     modifications.remove(insertions);
 
+    // moving a bool will not reset it's value, so to get the move out behaviour
```

Comment:
This should be done by the caller, not this. This function moves as an optimization and doesn't guarantee anything about the state of the object afterwards.